src/backend/mi_parse.ts

```diff
@@ -205,8 +205,10 @@ export function parseMI(output: string): MINode {
 		let oldContent = output;
 		let canBeValueList = output[0] == '[';
 		output = output.substr(1);
-		if (output[0] == '}' || output[0] == ']')
+		if (output[0] == '}' || output[0] == ']') {
+			output = output.substr(1); // ] or }
 			return [];
+		}
```

Comment:
uh any reason for this change? Could you include some test what this is supposed to fix or what didn't work in the tests file please

Reply:
I don't remember the exact command that caused it, but it was to do with an mi command returning {} followed by other data, where the other data was getting lost

Reply:
I've added an (artificial) test, which I can confirm did fail before this patch.